lib/index.ts

```diff
@@ -1,22 +1,65 @@
-import 'source-map-support/register';
-import {DepGraph, DepGraphBuilder, PkgInfo} from "@snyk/dep-graph";
+import "source-map-support/register";
+import { DepGraph, DepGraphBuilder, PkgInfo } from "@snyk/dep-graph";
 import * as os from "os";
 
-export function parseClojureDeps(name: string, version: string, depsEdnContent: string): DepGraph {
-  const rootPackage: PkgInfo = {name, version};
-  const depGraph = new DepGraphBuilder({name: 'clojure-deps'}, rootPackage);
+export function parseLein(
+  name: string,
+  version: string,
+  leinDepsTreeContent: string
+): DepGraph {
+  const parseLine = (line: string) =>
+    line
+      .replace("[", "")
+      .trimLeft()
+      .split(" ")
+      .slice(0, 2);
+  const calcDepth = (line: string) => (countSpaces(line) - 1) / 2;
+  return _parseTextTree(
+    name,
+    version,
+    leinDepsTreeContent,
+    calcDepth,
+    parseLine
+  );
+}
+
+export function parseClojureDeps(
+  name: string,
+  version: string,
+  depsEdnDepsTreeContent: string
+): DepGraph {
+  const parseLine = (line: string) => line.split(" ").slice(0, 2);
+  const calcDepth = (line: string) => countSpaces(line) / 2;
+  return _parseTextTree(
+    name,
+    version,
+    depsEdnDepsTreeContent,
+    calcDepth,
+    parseLine
+  );
+}
+
+function _parseTextTree(
```

Comment:
major changes in the PR are here - refactored `parseClojureDeps` out to this function, to allow usage in parsing both trees.